spi/src/main/java/org/eclipse/dataspaceconnector/spi/asset/Criterion.java

```diff
@@ -0,0 +1,63 @@
+package org.eclipse.dataspaceconnector.spi.asset;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+/**
+ * This class can be used to form select expressions e.g. in SQL statements. It is a way to express
+ * those statements in a generic way.
+ * For example:
+ * <p>
+ * {@code
+ * "operandLeft" = "name",
+ * "operator" = "=",
+ * "operandRight" = "someone"
+ * }
+ * <p>
+ * can be translated to [select * where name = someone]
+ * }
+ */
+public class Criterion {
```

Comment:
Just proposing:

- Left operand and right operand should implement a marker interface called `Operand`.
- Operator should implement a marker interface `Operator`. 

Enum would have the disadvantage that we are limited to a specific set of operators. Whereas interfaces are easily extendible, without touching the existing code base.

Using `String.Equals()` is not a good way to check the operator type. For example:
Some people could check for equals using the operator `=` or `==`. Which one is correct? Aren't these operators the same? 

In the policy model, we use marker interfaces, too.

Reply:
I should have explained this but a `Criterion` does not care whether you write "=" or "==" since its only a container. The object that evaluates the criterion - in the in-memory-index case thats the `CriterionToPredicateConverter` - needs to be aware of that. There could be converters that convert into SQL, Gremlin, etc, and that will really depend on the implementation of the index.
In the case of the InMemoryAssetIndex we're just converting to Predicates, and our "DSL" only accepts "=" as an operand - much like it was before.